## Урок 2. ДЗ (24.02.2021)
### Дискретные случайные величины. Закон распределения вероятностей.

### Биномиальный закон распределения. Распределение Пуассона

* [Теория](#theor)
* [Практическое задание. ДЗ.](#ex)
    * [Задание 1](#ex1)
    * [Задание 2](#ex2)
    * [Задание 3](#ex3)
    * [Задание 4](#ex4)

# <center>Теория <a class="anchor" id="theor"></a><center>

In [ ]:
АТС получает в среднем за час 480 вызовов. Определить вероятность того, что
за данную минуту она получит: ровно 3 вызова; от 2 до 5 вызовов.

**Случайная величина** — та, которая в результате испытания принимает только одно возможное значение.

Оно заранее не известно и может зависеть от множества случайных факторов. Случайные величины могут быть дискретными и непрерывными.

**Дискретная случайная величина** принимает отделенные друг от друга значения. Например, в результате стократного подбрасывания монетки орел может выпасть 50 или 51 раз (целое число в диапазоне от 0 до 100 включительно), но не 50 с половиной. В данном случае число выпаданий орла — пример дискретной случайной величины.

In [ ]:
import numpy as np

**Пример 1**

Смоделируем стократное подбрасывание монетки с помощью модуля **random** из библиотеки **numpy**. Результат будет дискретной случайной величиной — число выпаданий орла:

In [ ]:
# n - число испытаний
# p - вероятность появления события
np.random.binomial(n=100, p=0.5)

62

Клетку можно запускать несколько раз, число выпадений орла будет меняться.

**Пример 2**

Еще один пример дискретной случайной величины — число дождливых дней летом.

Летних дней всего 92, так что здесь мы можем получить случайное целое число из отрезка [0, 92]:

In [ ]:
np.random.randint(0, 93)

16

**Пример 3**

Есть также примеры дискретных величин без верхней границы. Например, число метеоритов, упавших на Землю за год, может принимать счетное множество значений от 0 до плюс бесконечности.

**Закон распределения вероятностей** дискретной случайной величины — это соответствие между возможными значениями этой величины и вероятностями, которые этим значениям соответствуют.

Закон распределения вероятностей можно представить в виде таблицы. Составим ее для количества выпаданий орла при трехкратном подбрасывании монеты:

<table border="3">
<tr><th>**X**</th><td>0</td><td>1</td><td>2</td><td>3</td></tr>
<tr><th>**p**</th><td>0.125</td><td>0.375</td><td>0.375</td><td>0.125</td></tr>
</table>

Из таблицы видно, что вероятность того, что при трехкратном подбрасывании монетки орел выпадет, к примеру, два раза, равна 0.375.

**Биномиальное распределение** — один из примеров дискретного распределения.

Биномиальный закон распределения — это закон распределения числа $X=k$ наступлений события $A$ в $n$ независимых испытаниях, в каждом из которых оно может произойти с вероятностью $p$.

Число наступления события — это дискретная величина из отрезка [0, $n$]. Вероятности возможных значений этой величины определяются по формуле Бернулли:

$$P_n(X=k) = C_n^k p^k q^{n-k}$$

где $p$ — это вероятность наступления события $A$ в $n$ независимых испытаниях, a $q = 1 - p$.

Математическое ожидание для биномиально распределенной дискретной величины можно вычислить по формуле:

$$M(X) = np$$

Дисперсия для биномиально распределенной дискретной величины вычисляется так:

$$D(X) = npq$$

**Пример 4**

Еще раз приведем таблицу вероятностей выпадания орла при трехкратном подбрасывании монеты. С помощью формулы Бернулли покажем, как вычисляются эти вероятности:

<table border="3">
<tr><th>**X**</th><td>0</td><td>1</td><td>2</td><td>3</td></tr>
<tr><th>**p**</th><td>0.125</td><td>0.375</td><td>0.375</td><td>0.125</td></tr>
</table>

$$p(k=1) = C_n^kp^kq^{n-k} = C_3^1p^1q^2 = \frac{3!}{1!(3-1)!} \cdot 0.5^1 \cdot 0.5^2 = 3 \cdot 0.5 \cdot 0.25 = 0.375$$


$$p(k=2) = C_n^kp^kq^{n-k} = C_3^2p^2q^1 = \frac{3!}{2!(3-2)!} \cdot 0.5^2 \cdot 0.5^1 = 3 \cdot 0.25 \cdot 0.5 = 0.375$$

$$p(k=3) = C_n^kp^kq^{n-k} = C_3^3p^3q^0 = \frac{3!}{3!(3-3)!} \cdot 0.5^3 \cdot 0.5^0 = 1 \cdot 0.125 \cdot 1 = 0.125$$

**Пример 5**

Будем находить математическое ожидание для биномиального распределения из предыдущего примера.

Параметр $n$ равняется 3, так как монету побрасывают три раза, а вероятность выпадания орла $p$ равняется 0.5.

Посчитаем математическое ожидание:

$$M(X) = np = 3 \cdot 0.5 = 1.5$$

Результат можно понимать так: в среднем при трехкратном подбрасывании монеты орел выпадает 1.5 раза.

Теперь посчитаем дисперсию, зная, что величина $q$ (вероятность невыпадания орла, то есть выпадания решки) равна также 0.5:

$$D(X) = npq = 3 \cdot 0.5 \cdot 0.5 = 0.75$$

**Распределение Пуассона**

Если проводится большое количество испытаний $n$ и при этом вероятность $p$ появления события $A$ в отдельном испытании мала, применяют формулу Пуассона для вычисления вероятности того, что событие произойдет $m$ раз в $n$ испытаниях:

$$P_m \approx \frac{\lambda^m}{m!}e^{-\lambda}$$

**где** $λ = np$.

Распределение Пуассона и формула Пуассона часто применяются в теории массового обслуживания (ТМО).

ТМО — это раздел теории вероятностей, в котором исследуется рациональный выбор структуры системы обслуживания и его процесса. В основе теории — изучение потоков требований на обслуживание, поступающих в систему и выходящих из нее, длительности ожидания и длины очередей.

**Пример 6**

Вероятность того, что среди писем, поступающих на определенный почтовый ящик, встретится письмо со спамом, составляет 0.01.

Найдите вероятность того, что среди 1000 писем, поступивших на него за месяц, будет 11 со спамом.

Применим формулу Пуассона, зная, что $m = 11$ и $λ = np = 1000 \cdot 0.01 = 10$:

$$P_{11} \approx \frac{\lambda^m}{m!}e^{-\lambda} \approx \frac{10^{11}}{11!}e^{-10} \approx  0.1137$$

Если использовать Python для расчетов, можно вычислить вероятность:

In [ ]:
n = 1000
m = 11
p = 0.01
lambda_ = n * p

(lambda_**m / np.math.factorial(m)) * np.exp(-lambda_)

0.1353352832366127

In [ ]:
from math import factorial

In [ ]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [ ]:
n = 144
k = 5
p = 1/2
X = combinations(n, k) * (p ** k) * (1 - p) ** (n - k)
X

2.1569168371633305e-35

In [ ]:
0.489 * 0.33 + 0.42 * 0.2975 + 0.09 * 0.6694

0.34656600000000004

In [ ]:
(7/10)*(6/9)*(2/11)*(1/10) + (7/10)*(3/9)*(9/11)*(2/10) + (7/10)*(3/9)*(2/11)*(9/10) + (3/10)*(7/9)*(9/11)*(2/10) + (3/10)*(7/9)*(2/11)*(9/10) + (3/10)*(2/9)*(9/11)*(8/10)

0.20484848484848484

In [ ]:
1 - 0.33 * 0.09

0.9703

# <center>Практическое задание <a class="anchor" id="ex"></a><center>

In [ ]:
from sympy import *
import numpy as np
from math import factorial as fac

def combinations(n, k):
    return int(fac(n) / (fac(k) * fac(n - k)))

def bern(p, n, k):
    return combinations(n,k) * p**(k) * (1 - p)**(n - k)

def puas(l,m):
    return l**m/fac(m)*np.exp(-l)

### Задание 1 <a class="anchor" id="ex1"></a>

Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

### Решение:

### Вероятность наступления события - попадание в цель 85 раз из 100 по формуле Бернулли.
### $$P_n(X=k) = C^k_np^kq^{n-k}$$
#### **где $p = 0.8$, а $q = 1-p = 0.2$**
### $$C^k_n = C^{85}_{100} = \frac{100!}{85!(100-85)!} =2.5333847134998864e+17$$
### $$P_{100}(X=85) = C^{85}_{100}p^{85}q^{100-85} = 0.04806 = 4.81\%$$

In [ ]:
print('C(85, 100) = ', combinations(100, 85))
print('p(0.8)^85 = ', 0.8**85)
print('q(0.2)^15 = ', 0.2**15)
print('P_bern(n=100, X=85) = ', combinations(100, 85)*(0.8**85)*(0.2**15))
print('P_bern(p=0.8, n=100, X=85) = ', bern(0.8, 100, 85))

C(85, 100) =  253338471349988640
p(0.8)^85 =  5.789604461865837e-09
q(0.2)^15 =  3.276800000000003e-11
P_bern(n=100, X=85) =  0.048061793700746556
P_bern(p=0.8, n=100, X=85) =  0.048061793700746355


### Вероятность наступления события - попадание в цель 85 раз из 100 по формуле Пуассона.
### $$P_m \approx \frac{\lambda^m}{m!}e^{-\lambda}$$
### **где $\lambda = pn,\,\,\, p = 0.8, \,\,\,n=100, \,\,\,m=85$**
### $$P_{85} \approx \frac{(0.8 \cdot 100)^{85}}{{85!}}e^{-(0.8 \cdot 100)} = 0.0371 = 3.71\%$$


In [ ]:
print('P_puas = ', (0.8*100)**85 / (fac(85)*exp(0.8*100)))
lda = 0.8*100
print('P_puas(lambda=p*n, m=85) = ', puas(lda, 85))

P_puas =  0.0370926143436918
P_puas(lambda=p*n, m=85) =  0.03709261434369179


### Задание 2 <a class="anchor" id="ex2"></a>

Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

### Решение:

### Определим вероятность наступления событий по формуле Пуассона.
### $$P_m \approx \frac{\lambda^m}{m!}e^{-\lambda}$$
### **где $\lambda = pn,\,\,\, p = 0.0004, \,\,\,n=5000, \,\,\,m=0$**, в первом случае (не перегорит ни одна лампочка) и $m=2$ во втором случае (перегорят ровно две)
### $$P_{0} \approx \frac{(2)^{0}}{{0!}}e^{-(2)} = 0.13533 = 13.533\%$$

### $$P_{2} \approx \frac{(2)^{2}}{{2!}}e^{-(2)} = 0.27067 = 27.067\%$$


### Определим вероятность наступления событий по формуле Бернулли.
### $$P_n(X=k) = C^k_np^kq^{n-k}$$
### $$p=0.0004,\,\,\,\,q=1-0.0004$$
### $$P_{5000}(X=0) = C^{0}_{5000}p^{0}q^{5000-0} = 0.13528 = 13.53\%$$
### $$P_{5000}(X=2) = C^{2}_{5000}p^{2}q^{5000-2} = 0.27073 = 27.07\%$$

In [ ]:
print('1) Не перегорит ни одна лампочка с вероятностью:')
lda = 0.0004*5000
print('P_puas(lambda=p*n, m=0) = ', puas(lda, 0))
print('P_bern(p, n, k) = ', bern(0.0004, 5000, 0))

1) Не перегорит ни одна лампочка с вероятностью:
P_puas(lambda=p*n, m=0) =  0.1353352832366127
P_bern(p, n, k) =  0.13528114551440706


In [ ]:
print('2) Перегорят две лампочки с вероятностью:')
lda = 0.0004*5000
print('P_puas(lambda=p*n, m=2) = ', puas(lda, 2))
print('P_bern(p, n, k) = ', bern(0.0004, 5000, 2))

2) Перегорят две лампочки с вероятностью:
P_puas(lambda=p*n, m=2) =  0.2706705664732254
P_bern(p, n, k) =  0.2707247150266753


### Задание 3 <a class="anchor" id="ex3"></a>

Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

### Решение:


### Вероятность наступления событий по формуле Бернулли.
### $$P_n(X=k) = C^k_np^kq^{n-k}$$
### $$p=0.5,\,\,\,\,q=1-0.5$$
### $$P_{144}(X=70) = C^{70}_{144}p^{70}q^{144-70} = 0.0628 = 6.28\%$$

In [ ]:
print('По формуле Бернулли 70 раз из 144 орел выпадет с вероятностью P(n=5000, X=2) = ', bern(0.5, 144, 70))


По формуле Бернулли 70 раз из 144 орел выпадет с вероятностью P(n=5000, X=2) =  0.06281178035144776


### Задание 4 <a class="anchor" id="ex4"></a>

В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

### Решение:

#### Вероятность того, что все мячи белые

### $$P(4) = \frac{7}{10} \cdot \frac{6}{9} \cdot \frac{9}{11} \cdot \frac{8}{10} = \frac{3024}{9900} = 0.305454 = 30,55\%$$

#### Посчитаем вероятность того, что два мяча белые - сумма вероятностей два белых из первого ящика и два черных из второго, два черных из первого и два белых из второго, один белый из первого и один белый из второго

### $$P(2) = \frac{7}{10} \cdot \frac{6}{9} \cdot \frac{2}{11} \cdot \frac{1}{10} + \frac{3}{10} \cdot \frac{2}{9} \cdot \frac{9}{11} \cdot \frac{8}{10} + \frac{7}{10} \cdot \frac{3}{9} \cdot \frac{9}{11} \cdot \frac{2}{10}= \frac{84 + 432 + 378}{9900} = 0.090303= 9,03\%$$

#### Посчитаем вероятность того, что хотя бы один мяч белый решив противоположную задачу как обратную к вероятности что все мячи черные

### $$P(1) = 1- \frac{3}{10} \cdot \frac{2}{9} \cdot \frac{2}{11} \cdot \frac{1}{10} = 1 - \frac{12}{9900} = 1 - 0.001212 = 0,998787 = 99,88\%$$

In [ ]:
a = 9900-12
b = 10*9*11*10
print(a, b, a/b)

9888 9900 0.9987878787878788
